app/jobs/order_cycle_opening_job.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+# Trigger jobs for any order cycles that recently opened
+class OrderCycleOpeningJob < ApplicationJob
+  def perform
+    recently_opened_order_cycles.each do |oc_id|
+      SubscriptionPlacementJob.perform_later(oc_id)
+    end
+  end
```

Comment:
I imagined the timestamp to be set here. It's like the initial event as been dispatched. It can be done in bulk and it prevents more jobs from being enqueued which then cancel themselves because they have been executed already.

Reply:
I think I'd like to discuss this again, as I'm not sure about it. 

I find myself wondering if indeed we do need an intermediate job (OrderCycleWebhookJob) simply for creating a tiny data blob (payload)  and scheduling for delivery. 
Yes, perhaps if there were hundreds of order cycles with webhooks opening at the same time, it might take a bit longer. And maybe if having instant notifications was crucial, this could pose a problem. But I don't think we're anywhere near the need for that optimisation. 

So I think I would move that logic into OrderCycleOpeningJob. Let's discuss when we're both available.